playwright/pages/bot.ts

```diff
@@ -179,46 +183,39 @@ export class Bot extends Client {
                 await cli.initRustCrypto({ useIndexedDB: false });
                 cli.setGlobalErrorOnUnknownDevices(false);
                 await cli.startClient();
-
-                if (opts.bootstrapCrossSigning) {
-                    // XXX: workaround https://github.com/element-hq/element-web/issues/26755
-                    //   wait for out device list to be available, as a proxy for the device keys having been uploaded.
-                    await cli.getCrypto()!.getUserDeviceInfo([credentials.userId]);
-
-                    await cli.getCrypto()!.bootstrapCrossSigning({
-                        authUploadDeviceSigningKeys: async (func) => {
-                            await func({
-                                type: "m.login.password",
-                                identifier: {
-                                    type: "m.id.user",
-                                    user: credentials.userId,
-                                },
-                                password: credentials.password,
-                            });
-                        },
-                    });
-                }
-
-                if (opts.bootstrapSecretStorage) {
-                    const passphrase = "new passphrase";
-                    const recoveryKey = await cli.getCrypto().createRecoveryKeyFromPassphrase(passphrase);
-                    Object.assign(cli, { __playwright_recovery_key: recoveryKey });
-
-                    await cli.getCrypto()!.bootstrapSecretStorage({
-                        setupNewSecretStorage: true,
-                        setupNewKeyBackup: true,
-                        createSecretStorageKey: () => Promise.resolve(recoveryKey),
-                    });
-                }
-
                 return cli;
             },
             {
                 homeserver: this.homeserver.config,
-                credentials: await this.getCredentials(),
+                credentials,
                 opts: this.opts,
             },
         );
-        return this.handlePromise;
+
+        if (this.opts.bootstrapCrossSigning) {
+            // XXX: workaround https://github.com/element-hq/element-web/issues/26755
+            //   wait for out device list to be available, as a proxy for the device keys having been uploaded.
+            await clientHandle.evaluate(async (cli, credentials) => {
+                await cli.getCrypto()!.getUserDeviceInfo([credentials.userId]);
```

Comment:
The test fails because of this condition: https://github.com/element-hq/matrix-react-sdk/blob/c2945177c91fb9b961d6e013fc21a269fb13676e/playwright/pages/bot.ts#L179
Basically the client is returned without starting it.

Reply:
ah thanks